avocado/core/runner.py

```diff
@@ -468,6 +469,12 @@ def _iter_variants(template, mux):
         :raises ValueError: When variant and template declare params.
         """
         for variant, params in mux.itertests():
+            variant_name = ''
+            for x in params[0]:
+                path = x.path[len('/run'):]
+                variant_name += path.replace('/', '-')
+            # trim first '-'
+            variant_name = variant_name[1:]
```

Comment:
Btw implementation-wise if we chose to split variant-id to variant-id and variant-name I'd prefer keeping the API and using:

```python
class VariantId(object):
    def __init__(self, variant_id, name):
        self.id = variant_id
        self.name = name
    def __str__(self):
        return self.id
```
which is backward compatible and the new avocado would simply ask whether the variant has `name` attribute and if not it'd use `str(variant)` instead.